hub/src/main/java/ch/prevo/open/hub/match/MatcherService.java

```diff
@@ -29,7 +34,12 @@
     }
 
     private Optional<InsurantInformation> findMatchingEntry(Set<InsurantInformation> retirementFundEntries, InsurantInformation termination) {
-        return retirementFundEntries.stream().filter(commencement -> isMatching(commencement, termination)).findAny();
+        final List<InsurantInformation> result = retirementFundEntries.stream().filter(commencement -> isMatching(commencement, termination)).collect(Collectors.toList());
+        if (result.size() > 1) {
```

Comment:
Couldn't you use `count()` here on the stream so you don't have to collect it and stream it again when returning `findAny()`

Reply:
Actually I tried that initially, but it turned out that count() is already a terminal operation and the stream cannot be used afterwards. Java streams do not support forking, thus one has to buffer them in order to be able to reuse them.